<launch>
    <!-- 
    Parameters files for :
        - Distance server
        - Agent distance publisher
    -->
    <arg name="occupancy_preproc_config_file" default="$(find ros_explore_mapping)/config/occupancy_preproc/occ_preproc_config.yaml"/>
    <arg name="dist_server_config_file" default="$(find ros_explore_mapping)/config/distance_mapping/dist_server_standalone_config.yaml"/>
    <arg name="agent_dist_publisher_config_file" default="$(find ros_explore_mapping)/config/distance_mapping/agent_dist_publisher_config.yaml"/>
    
    <!-- Occupancy preprocessing -->
    <node name="occupancy_preprocessing" pkg="ros_explore_mapping" type="occupancy_preproc_server.py" output="screen">
        <rosparam file="$(arg occupancy_preproc_config_file)"/>
    </node>
    <!-- Distance server -->
    <node pkg="ros_explore_mapping" type="distance_map_server.py" name="dist_server" output="screen">
        <rosparam file="$(arg dist_server_config_file)"/>
    </node>
    <!-- Agent distance publisher -->
    <node pkg="ros_explore_mapping" type="agent_dist_publisher.py" name="agent_dist_pub" output="screen">
        <rosparam file="$(arg agent_dist_publisher_config_file)"/>
    </node>
</launch>
